### PR TITLE
`SoB/BDKWallet`: Add `Fidelity`, `SwapCoin` variants to `KeychainKind` enum

### PR DESCRIPTION
**Pull request by wthrajat**
Aims to fix: #155

### CODE REVIEW

File: src/wallet/bdk.rs [hunk 1]
```diff
@@ -82,15 +82,21 @@ pub struct Wallet {
 /// Each kind represents an unhardened index value. Starting with External = 0.
 #[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Hash, Clone, Copy)]
 pub enum KeychainKind {
-    External = 0isize,
+    External,
     Internal,
+    Fidelity,
+    SwapCoin { id: u32 },
```
Comment (mojoX911) [inline src/wallet/bdk.rs:88]: Whats this id? Why is it needed?

File: src/wallet/bdk.rs [hunk 2]
```diff
@@ -82,15 +82,21 @@ pub struct Wallet {
...
 }
 
 impl KeychainKind {
     fn index_num(&self) -> u32 {
         match self {
             Self::External => 0,
             Self::Internal => 1,
+            Self::Fidelity => 2,
+            Self::SwapCoin { id } => 3 + id,
+            Self::Contract => 4,
```
Comment (mojoX911) [inline src/wallet/bdk.rs:99]: This won't work. When id=1, SwapCoin and Contract keychain index will be same. This will mess things up.

**Review by mojoX911 — APPROVED** [review]
ACK.

**Review by Shourya742 — APPROVED** [review]
ACK

**Pull request merged**